addons/binding/org.openhab.binding.xiaomivacuum/ESH-INF/thing/vacuumThing.xml

```diff
@@ -0,0 +1,225 @@
+		</channels>
+	</channel-group-type>
+	<channel-group-type id="dnd">
+		<label>Do not Disturb</label>
+		<channels>
+			<channel id="dnd_function" typeId="dnd_function" />
+			<channel id="dnd_start" typeId="dnd_start" />
+			<channel id="dnd_end" typeId="dnd_end" />
+		</channels>
+	</channel-group-type>
+	<channel-group-type id="history">
+		<label>History</label>
+		<channels>
+			<channel id="total_clean_area" typeId="total_clean_area" />
+			<channel id="total_clean_time" typeId="total_clean_time" />
+			<channel id="total_clean_count" typeId="total_clean_count" />
+		</channels>
+	</channel-group-type>
+
+	<!-- Status channels -->
+	<channel-type id="clean_area">
+		<item-type>Number</item-type>
+		<label>Cleaning Area</label>
+		<state pattern="%.2d M2" readOnly="true" />
+	</channel-type>
+	<channel-type id="clean_time">
+		<item-type>Number</item-type>
+		<label>Cleaning Time</label>
+		<state pattern="%.2d min" readOnly="true" />
+	</channel-type>
+	<channel-type id="dnd_enabled">
+		<item-type>Switch</item-type>
+		<label>Do not Disturb</label>
+	</channel-type>
+	<channel-type id="error_code">
+		<item-type>String</item-type>
+		<label>Error</label>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="fan_power">
+		<item-type>Number</item-type>
+		<label>Fan Power</label>
+		<state pattern="%.0d %%" readOnly="true" />
+	</channel-type>
+	<channel-type id="in_cleaning">
+		<item-type>Number</item-type>
+		<label>in_cleaning</label>
+		<state pattern="%.0d" readOnly="true" />
+	</channel-type>
+	<channel-type id="map_present" advanced="true">
+		<item-type>Number</item-type>
+		<label>map_present</label>
+		<category>Energy</category>
+		<state pattern="%.0d" readOnly="true" />
+	</channel-type>
+	<channel-type id="msg_seq" advanced="true">
+		<item-type>Number</item-type>
+		<label>msg_seq</label>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="msg_ver" advanced="true">
+		<item-type>Number</item-type>
+		<label>msg_ver</label>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="state">
+		<item-type>String</item-type>
+		<label>state</label>
+		<state readOnly="true" />
+	</channel-type>
+
+	<!-- Consumables channels -->
+	<channel-type id="main_brush_time">
+		<item-type>Number</item-type>
+		<label>Main Brush Time</label>
+		<state pattern="%.2d min" readOnly="true" />
+	</channel-type>
+	<channel-type id="side_brush_time">
+		<item-type>Number</item-type>
+		<label>Side Brush Time</label>
+		<state pattern="%.2d min" readOnly="true" />
+	</channel-type>
+	<channel-type id="filter_time">
+		<item-type>Number</item-type>
+		<label>Filter Usage Time</label>
+		<state pattern="%.2d min" readOnly="true" />
+	</channel-type>
+	<channel-type id="sensor_dirt_time">
+		<item-type>Number</item-type>
+		<label>Sensor Dirt Time</label>
+		<state pattern="%.2d min" readOnly="true" />
+	</channel-type>
+
+	<!-- actions -->
+	<channel-type id="vacuum">
```

Comment:
IMHO having all these individual channels is redundant, the "commands" one should suffice. 
If it defines a set of command options like e.g. this (you could use numeric values too, if you like these better):

```xml
<channel-type id="commands">
	<item-type>String</item-type>
	<label>Execute Command</label>
	<state>
		<options>
			<option value="vacuum">Vacuum</option>
			<option value="clean">Clean</option>
			<option value="pause">Pause</option>
			<option value="dock">Dock</option>
		</options>
	</state>
</channel-type>
```

then users can define their sitemaps like this:

```
Switch item=xiaomi_commands mappings=[vacuum="Vacuum", pause="Pause", dock="Dock"]
```

and will get nice little buttons like this:
![image](https://user-images.githubusercontent.com/2592443/28201205-3e267e24-6870-11e7-9fb3-fb7a98703eb6.png)

Then you don't need all those individual switch channels, where some have two states and even only have one, but all somehow are interconnected because in fact the robot can do only one thing at a time like vaccum/spot_clean/pause/dock. What do you think?